tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/cluster/warnings/ClusterCleanNodeShutdownTest.java

```diff
@@ -0,0 +1,108 @@
+
+import java.lang.invoke.MethodHandles;
+import java.util.Collection;
+
+import org.apache.activemq.artemis.api.core.SimpleString;
+import org.apache.activemq.artemis.core.postoffice.Binding;
+import org.apache.activemq.artemis.core.server.cluster.impl.MessageLoadBalancingType;
+import org.apache.activemq.artemis.logs.AssertionLoggerHandler;
+import org.apache.activemq.artemis.tests.integration.cluster.distribution.ClusterTestBase;
+import org.junit.Assert;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ClusterCleanNodeShutdownTest extends ClusterTestBase {
+
+   private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+   @Test
+   public void testNoWarningErrorsDuringRestartingNodesInCluster() throws Exception {
+      setupServer(0, isFileStorage(), isNetty());
+      setupServer(1, isFileStorage(), isNetty());
+
+      setupClusterConnection("cluster0", "queues", MessageLoadBalancingType.ON_DEMAND, 1, isNetty(), 0, 1);
+
+      setupClusterConnection("cluster1", "queues", MessageLoadBalancingType.ON_DEMAND, 1, isNetty(), 1, 0);
+
+      startServers(0, 1);
+      Thread.sleep(2000);
+
+      logger.debug("server 0 = {}", getServer(0).getNodeID());
+      logger.debug("server 1 = {}", getServer(1).getNodeID());
+
+      setupSessionFactory(0, isNetty(), 15);
+      setupSessionFactory(1, isNetty());
+
+      // now create the 2 queues and make sure they are durable
+      createQueue(0, "queues.testaddress", "queue10", null, true);
+      createQueue(1, "queues.testaddress", "queue10", null, true);
+
+      addConsumer(0, 0, "queue10", null);
+
+      waitForBindings(0, "queues.testaddress", 1, 1, true);
+      waitForBindings(1, "queues.testaddress", 1, 0, true);
+
+      waitForBindings(0, "queues.testaddress", 1, 0, false);
+      waitForBindings(1, "queues.testaddress", 1, 1, false);
+
+      printBindings(2);
+
+      sendInRange(1, "queues.testaddress", 0, 10, true, null);
+
+      try (AssertionLoggerHandler loggerHandler = new AssertionLoggerHandler(true)) {
+         logger.info("*****************************************************************************");
+         stopServers(0);
+         // Waiting some time after stopped
+         Thread.sleep(2000);
+         logger.info("*****************************************************************************");
+         Assert.assertFalse("Connection failure detected for an expected DISCONNECT event",  loggerHandler.findText("AMQ212037", " [code=DISCONNECTED]"));
+         Assert.assertFalse("WARN found",loggerHandler.hasLevel(AssertionLoggerHandler.LogLevel.WARN));
+      }
+      startServers(0);
+
+      waitForBindings(0, "queues.testaddress", 1, 1, true);
+      waitForBindings(1, "queues.testaddress", 1, 0, true);
+
+      waitForBindings(0, "queues.testaddress", 1, 0, false);
+      waitForBindings(1, "queues.testaddress", 1, 1, false);
+
+      printBindings(2);
```

Comment:
do you always need the printBindings? Can't you add it only when something "bad" happened?

or at the worst case... if (logger.isDebugEnabled()) ?